js/ui/search.js

```diff
@@ -367,14 +394,35 @@ const GridSearchResults = new Lang.Class({
 });
 Signals.addSignalMethods(GridSearchResults.prototype);
 
+const SearchResultsBin = new Lang.Class({
+    Name: 'SearchResultsBin',
+    Extends: St.BoxLayout,
+
+    vfunc_allocate: function(box, flags) {
+        let themeNode = this.get_theme_node();
+        let maxWidth = themeNode.get_max_width();
+        let availWidth = box.x2 - box.x1;
+        let adjustedBox = box;
+
+        if (availWidth > maxWidth) {
+            let excessWidth = availWidth - maxWidth;
+            adjustedBox.x1 += Math.floor(excessWidth / 2);
+            adjustedBox.x2 -= Math.floor(excessWidth / 2);
+        }
+
+        this.parent(adjustedBox, flags);
+    }
+});
+
 const SearchResults = new Lang.Class({
     Name: 'SearchResults',
 
     _init: function() {
-        this.actor = new St.BoxLayout({ name: 'searchResults',
-                                        vertical: true });
+        this.actor = new SearchResultsBin({ name: 'searchResults',
+                                            vertical: true });
+        Util.blockClickEventsOnActor(this.actor);
 
-        this._content = new St.BoxLayout({ name: 'searchResultsContent',
+        this._content = new St.BoxLayout({ name: 'searchResultsBox',
```

Comment:
The name of this actor is changed, yet nothing in the theme reflects the new name. I believe the change is accidental and would not be surprised if it caused some of the issues mentioned in the commit message.